### PR TITLE
Replace metadata visitors in accessor generation

### PR DESCRIPTION
**Pull request by jbartok**
Adapt accessors generation code to use the `kotlinx-metadata` node API instead of the visitor API that is deprecated in the next version.

### CODE REVIEW

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test RFN

**Review by eskatos — APPROVED** [review]
:+1: Looks good!

I left a few minor comments, this can be merged once addressed.

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Comment by jbartok** [comment]
@bot-gradle cancel

File: subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/normalization/KotlinApiClassExtractor.kt [hunk 1]
```diff
@@ -125,13 +125,13 @@ class KotlinApiMemberWriter(apiMemberAdapter: ClassVisitor) : ApiMemberWriter(ap
...
-        var extraString: String? = null
-        var packageName: String? = null
-        var extraInt: Int? = null
+        var kind = 0
+        var metadataVersion = IntArray(0)
+        var data1 = arrayOf<String>()
+        var data2 = arrayOf<String>()
+        var extraString = ""
+        var packageName = ""
+        var extraInt = 0
```
[the remark this answers is not in the document]
Comment (jbartok) [reply subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/normalization/KotlinApiClassExtractor.kt:134]: Indeed, this is a change specific to bumping the library version. Moving it to that PR. 

PS: the default values aren't arbitrary.

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Pull request merged**